central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -84,88 +93,169 @@ type ResultEmail struct {
-		profiles += "..."
-	}
-
 	formatEmailSub := &formatSubject{
 		BrandedPrefix: branding.GetCombinedProductAndShortName(),
 		ScanConfig:    req.ScanConfigName,
-		Profiles:      len(profiles),
+		Profiles:      len(req.Profiles),
 	}
 
 	log.Infof("Sending email for scan config %s", req.ScanConfigName)
 	go rg.sendEmail(req.Ctx, zipData, formatEmailBody, formatEmailSub, req.Notifiers)
 }
 
-// getDataForReport returns map of cluster id and
-func (rg *complianceReportGeneratorImpl) getDataForReport(req *ComplianceReportRequest) (*ResultEmail, error) {
-	// TODO ROX-24356: Implement query to get checkresults data to generate cvs for compliance reporting
+// getDataforReport returns map of cluster id and slice of ResultRow
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest) *ResultEmail {
+	clusters := req.ClusterIDs
+	resultsCSV := make(map[string][]*ResultRow)
+	resultEmailComplianceReport := &ResultEmail{
+		TotalPass:  0,
+		TotalMixed: 0,
+		TotalFail:  0,
+		Clusters:   0,
+	}
+
+	for _, clusterID := range clusters {
+		parsedQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfig, req.ScanConfigID).
+			AddExactMatches(search.ClusterID, clusterID).
+			ProtoQuery()
+		resultCluster := []*ResultRow{}
+
+		err := rg.checkResultsDS.WalkByQuery(req.Ctx, parsedQuery, func(checkResult *storage.ComplianceOperatorCheckResultV2) error {
+			row := &ResultRow{
+				ClusterName: checkResult.GetClusterName(),
+				CheckName:   checkResult.GetCheckName(),
+				Description: checkResult.GetDescription(),
+				Status:      checkResult.GetStatus().String(),
+			}
+			// get profile name for the check result
+			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, checkResult.GetScanRefId()).ProtoQuery()
+			profiles, err := rg.profileDS.SearchProfiles(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(profiles) < 1 {
+				row.Profile = "Data not available"
+				log.Errorf("profile not found for cluster %s and check name %s", clusterID, checkResult.GetCheckName())
+			} else {
+				row.Profile = fmt.Sprintf("%s %s", profiles[0].GetName(), profiles[0].GetProfileVersion())
+			}
+
+			// get remediation for the check result
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, checkResult.GetCheckName()).AddExactMatches(search.ClusterID, checkResult.GetClusterId()).ProtoQuery()
+			remediations, err := rg.remediationDS.SearchRemediations(req.Ctx, q)
+			log.Infof("Remediation is %v", remediations)
+			if err != nil {
+				row.Remediation = "Data not available"
+				log.Errorf("remediations not found for cluster %s and check name %s. Error returned %s", clusterID, checkResult.GetCheckName(), err)
+			} else if len(remediations) == 0 {
+				row.Remediation = "No remediation available"
+				log.Errorf("remediations not found for cluster %s and check name %s", clusterID, checkResult.GetCheckName())
```

Comment:
no need to log this.  Most checks will not have a remediation especially if they pass.  So no remediation is the most common case.

In fact I could be convinced to simply leave `row.Remediation = ""` in this case.  But I'm OK with `No remediation available` too.